src/lib/isLength.js

```diff
@@ -5,15 +5,39 @@ export default function isLength(str, options) {
   assertString(str);
   let min;
   let max;
+  let exact;
+  let result;
+  let isValid = false;
```

Comment:
I think most "buffer" variables here are unnecessary and add to indirection. 

```suggestion
  // accessing 'arguments' for backwards compatibility: isLength(str, min [, max])
  const min =
    typeof options === 'object' ? options.min || 0 : arguments[1] || 0;
  const max = typeof options === 'object' ? options.max : arguments[2];
```